plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,305 @@
+                           Map<String, String> join,
+                           List<String> remove,
+                           List<String> trim,
+                           List<String> uppercase,
+                           List<String> lowercase) {
+        this.update = update;
+        this.rename = rename;
+        this.convert = convert;
+        this.split = split;
+        this.gsub = gsub;
+        this.join = join;
+        this.remove = remove;
+        this.trim = trim;
+        this.uppercase = uppercase;
+        this.lowercase = lowercase;
+    }
+
+    public Map<String, Object> getUpdate() {
+        return update;
+    }
+
+    public Map<String, String> getRename() {
+        return rename;
+    }
+
+    public Map<String, String> getConvert() {
+        return convert;
+    }
+
+    public Map<String, String> getSplit() {
+        return split;
+    }
+
+    public Map<String, Tuple<Pattern, String>> getGsub() {
+        return gsub;
+    }
+
+    public Map<String, String> getJoin() {
+        return join;
+    }
+
+    public List<String> getRemove() {
+        return remove;
+    }
+
+    public List<String> getTrim() {
+        return trim;
+    }
+
+    public List<String> getUppercase() {
+        return uppercase;
+    }
+
+    public List<String> getLowercase() {
+        return lowercase;
+    }
+
+    @Override
+    public void execute(Data data) {
+        if (update != null) { doUpdate(data); }
+        if (rename != null) { doRename(data); }
+        if (convert != null) { doConvert(data); }
+        if (split != null) { doSplit(data); }
+        if (gsub != null) { doGsub(data); }
+        if (join != null) { doJoin(data); }
+        if (remove != null) { doRemove(data); }
+        if (trim != null) { doTrim(data); }
+        if (uppercase != null) { doUppercase(data); }
+        if (lowercase != null) { doLowercase(data); }
+    }
+
+    private void doUpdate(Data data) {
+        for(Map.Entry<String, Object> entry : update.entrySet()) {
+            data.addField(entry.getKey(), entry.getValue());
+        }
+    }
+
+    private void doRename(Data data) {
+        for(Map.Entry<String, String> entry : rename.entrySet()) {
+            if (data.containsProperty(entry.getKey())) {
+                Object oldVal = data.getProperty(entry.getKey());
+                data.getDocument().remove(entry.getKey());
+                data.addField(entry.getValue(), oldVal);
+            }
+        }
+    }
+
+    private Object parseValueAsType(Object oldVal, String toType) {
+        switch (toType) {
+            case "integer":
+                oldVal = Integer.parseInt(oldVal.toString());
+                break;
+            case "float":
+                oldVal = Float.parseFloat(oldVal.toString());
+                break;
+            case "string":
+                oldVal = oldVal.toString();
+                break;
+            case "boolean":
+                String val = oldVal.toString();
```

Comment:
maybe use `Booleans#parseBoolean(...)` here? (even though it is outside the ingest package, we can add a todo here mentioning that we may need to fork it later on when moving ingest to a dedicated library)